### ISSUE TITLE
Mobile site root route.

### ISSUE THREAD

**Issue by eicca** [onaseef/yomobi#441]
Since `locomotivecms` uses catch-all route for pages there is a problems with this route: https://github.com/onaseef/yomobi/blob/master/config/routes.rb#L118

Perhaps there is a better solution than add all `locomotive` routes into `exceptions` array?
Or have you really need this behavior? Because you have the same routes with mobile domain.

**Comment by onaseef** [comment]
What problem are you seeing with the route mobile#index?  

I will ask the previous developer to comment on this issue. He will have a better idea of a solution.

**Comment by eicca** [comment]
`get ':company'` means: grab all urls like: `yomobi.com/:pattern`. And in the same time we have pages with dynamic slugs and they has urls like: `yomobi.com/about-us`, `yomobi.com/blog` etc.

So with `get ':company'` when we navigate to `yomobi.com/about-us` then we get mobile site with `db_name == about-us` instead of locomotive page .

**Comment by onaseef** [comment]
What do recommend as the best technical solution? 

We would need to make sure any solution we choose won't affect future mobile site URLs. Maybe we need to create a unique reserved URL path for yomobi pages. For example: yomobi.com/p/about-us & yomobi.com/p/blog. 

That way we can create any page name we want and not worry that someone might have created a mobile URL with the same name already. 

What do you think of this approach? I  am open to a better solution

**Comment by eicca** [comment]
`yomobi.com/p/about-us` & `yomobi.com/p/blog`  - is a good solution, but as far as I know it's not good for SEO purposes, because these pages becomes 'deeper'.

I don't understand, why you need `get ':company'` route if you already have exactly same page with sub-domain https://github.com/onaseef/yomobi/blob/master/config/routes.rb#L6?

And if this route is needed, I think it will be more clear to make subdirectory for him. For example: `yomobi.com/mobile/:company`

**Comment by onaseef** [comment]
We need to handle the case of yomobi.com/company because we have numerous sites that added the auto-redirect script in that format. Also, the social media buttons at the bottom of the mobile site use the format because Facebook doesn't seem to handle company.yomobi.com properly. If we can find a good workaround we can remove the yomobi.com/company route and just use the subdominant format. But we need to be backwards compatible for the 60 sites using the current format. We could create an array to handle sites using the old route format. 

What do you think is the best way to proceed?

**Comment by eicca** [comment]
I think the most right solution is to solve problem with facebook. Unfortunately I never met this kind of problem before, so I can't advice something about it.

I'll try investigate this problem later, but as dirty and quick solution we can share link to `yomobi.com/mobile/:company` that redirects user to `company.yomobi.com`.

As for existing companies - yes, we can move them into exception array and send email with new redirection link. And then remove that array in a few months.

**Comment by gilbert** [comment]
I agree with Mikhail. Transitioning to subdomains-only is probably best overall. The exception list for existing companies can redirect to each appropriate subdomain.

I don't know anything about the facebook problem, unfortunately.

**Comment by onaseef** [comment]
What is the status of this issue?

Also, rather than yomobi.com/mobile/:company, I'd prefer yomobi.com/m/:company.

**Comment by eicca** [comment]
Probably I know how to fix FB problem. We need to remove `www` from link with subdomain.

For example in http://stimfg.yomobi.com/ there are share link http://www.facebook.com/share.php?u=http%3A%2F%2Fwww.yomobi.com%2Fstimfg&t=STI%20Manufacturing%20Inc.%27s%20Mobile%20Website%20-%20YoMobi

If we change it to subdomain version it's still would work: http://www.facebook.com/share.php?u=http%3A%2F%2Fstimfg.yomobi.com&t=STI%20Manufacturing%20Inc.%27s%20Mobile%20Website%20-%20YoMobi

**Comment by eicca** [comment]
@onaseef if you can confirm that it's works, I can implement this changes:
1. remove `/:company` routes
2. add exception array to existing companies
3. change FB share link
4. change redirect script?
5. something else?

**Comment by onaseef** [comment]
Gilbert, I think you confirmed this would work without issues? Is that correct? 

 Mikhael, when Gilbert confirms there are no issues, you can move forward with this change.

Notes:
1. Do you know how to get the list of existing companies or do I need to do this? 
2. The redirect script is already in the correct format
3. Gilbert, can you think of anything else?

**Comment by faust45** [comment]
Sorry guys, but fixing issues in app is not part of our job, we discuss only run CMS.

**Comment by onaseef** [comment]
If you read through the notes, you will see that this is not a bug with the application. This is a bug created when you installed the CMS. After the CMS was installed, all of the mobile site paths that were previously working are now broken. What needs to be fixed is the integration of the CMS into the application in a way that does not break the exiting functionality.

**Comment by eicca** [comment]
@onaseef, about (2): I can get them when you give me access to the current production (during migration to cedar stack).
Ok, Im waiting for @mindeavor reply.

**Comment by gilbert** [comment]
As long as the `:mobile` path is updated in routes.rb, I think everything should be good.

**Comment by eicca** [comment]
I've added `OLD_SITES_URLS` to `environment.rb`. Looks like it works.

**Comment by onaseef** [comment]
Thank you. Would you please check in the most recent work into github so I can check It later tonight. Thank you.

**Comment by eicca** [comment]
Ok, I pushed the latest changes: https://github.com/onaseef/yomobi/commits/rails-3.2
